False positive in rule "vue/no-dupe-keys"

<!--
    This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

    1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
    2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes

    Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**

* **ESLint Version:** 4.14.0
* **eslint-plugin-vue Version:** 4.0.0
* **Node Version:** 8.5.0

**Please show your full configuration:**
```js
// basic configuration
module.exports = {
  root: true,
  extends: [
    'airbnb-base',
    'plugin:vue/recommended',
  ],
  parserOptions: {
    parser: 'babel-eslint'
  },
  plugins: ['vue']
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
// @vue/component
export const compA = {
  props: {
    propA: String,
  },
};

// @vue/component
export const compB = {
  props: {
    propA: Boolean, / <-- error line
  },
};
```

**What did you expect to happen?**
Should not raise no-dupe-keys error

**What actually happened? Please include the actual, raw output from ESLint.**
As you can see i export two components in one file. Both components have one prop with the same name and plugin is warning me about duplicated propA.

```
C:\dev\foobar\src\app.js
  11:5  error  Duplicated key 'propA'  vue/no-dupe-keys

✖ 1 problem (1 error, 0 warnings)
```

--- a/lib/rules/no-dupe-keys.js
+++ b/lib/rules/no-dupe-keys.js
@@ -13,8 +13,6 @@ const utils = require('../utils')
 const GROUP_NAMES = ['props', 'computed', 'data', 'methods']
 
 function create (context) {
-  const usedNames = []
-
   const options = context.options[0] || {}
   const groups = new Set(GROUP_NAMES.concat(options.groups || []))
 
@@ -23,7 +21,9 @@ function create (context) {
   // ----------------------------------------------------------------------
 
   return utils.executeOnVue(context, (obj) => {
+    const usedNames = []
     const properties = utils.iterateProperties(obj, groups)
+
     for (const o of properties) {
       if (usedNames.indexOf(o.name) !== -1) {
         context.report({
@@ -34,6 +34,7 @@ function create (context) {
           }
         })
       }
+
       usedNames.push(o.name)
     }
   })

--- a/tests/lib/rules/no-dupe-keys.js
+++ b/tests/lib/rules/no-dupe-keys.js
@@ -80,6 +80,26 @@ ruleTester.run('no-dupe-keys', rule, {
         }
       `,
       parserOptions: { ecmaVersion: 8, sourceType: 'module', ecmaFeatures: { experimentalObjectRestSpread: true }}
+    },
+
+    {
+      filename: 'test.js',
+      code: `
+        // @vue/component
+        export const compA = {
+          props: {
+            propA: String
+          }
+        }
+
+        // @vue/component
+        export const compB = {
+          props: {
+            propA: String
+          }
+        }
+      `,
+      parserOptions: { ecmaVersion: 6, sourceType: 'module' }
     }
   ],
 

